tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/object/test_bir_negative.bal

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import bir/objs;
+
+function f1(objs:Bar bar) {
+    _ = bar is objs:Foo;
+}
+
+function f2(objs:Foo foo) {
+    _ = foo is objs:Bar;
+}
+
+function f3(objs:Xyz xyz) {
+    _ = xyz is objs:Qux;
+}
+
+function f4(objs:Qux qux) {
+    _ = qux is objs:Xyz;
+}
```

Comment:
How about also testing with an object defined in the current module vs one from the imported module, where the qualifiers are incompatible?

Reply:
Can add positive tests also.